code/modules/client/preferences/species_features/basic.dm

```diff
@@ -77,7 +94,7 @@
 	return data
 
 /datum/preference/color/facial_hair_color
-	priority = PREFERENCE_PRIORITY_BODYPARTS
+	priority = PREFERENCE_PRORITY_LATE_BODY_TYPE // Need to happen after hair oclor is set so we can match by default
```

Comment:
there should just be a toggle button to enable/disable matching the hairstyle color